engine/src/main/java/org/hibernate/search/engine/cfg/impl/ConvertUtils.java

```diff
@@ -112,6 +113,23 @@ else if ( "true".equalsIgnoreCase( cleaned ) ) {
 				.map( Long::parseLong );
 	}
 
+	public static Optional<BeanReference> convertBeanReference(Class<?> expectedType, Object value) {
+		if ( expectedType.isInstance( value ) ) {
+			return Optional.of( BeanReference.ofInstance( expectedType.cast( value ) ) );
+		}
+		if ( value instanceof BeanReference ) {
+			return Optional.of( (BeanReference) value );
+		}
+		if ( value instanceof Class ) {
+			return Optional.of( BeanReference.ofType( (Class<?>) value ) );
+		}
+		if ( value instanceof String ) {
+			return optionalTrimmedNonEmpty( (String) value )
+					.map( BeanReference::ofName );
+		}
+		throw log.invalidBeanReferencePropertyValue( expectedType );
```

Comment:
Use a string containing the fully-qualified class name. This will be interpreted as a name reference, and the reflection bean resolver (the default, used as a fallback even when using Spring/CDI) will interpret names as FQCN.

We test that it works, albeit indirectly. In `backend-tck-analysis-custom.properties` for example we use a FQCN as a value for a configuration property.

Reply:
OK, cool.